### ISSUE TITLE
Configure modules in ecosystem.config.js

### ISSUE THREAD

**Issue by nitrocode** [Unitech/pm2#3483]
Similar to https://github.com/Unitech/pm2/issues/3268

## What's going wrong?

New changes now allow us to install pm2 modules in a `dependencies` key. I'd like to also be able to configure the dependencies directly from the `ecosystem.config.js` to avoid manually running configuration commands.

## How could we reproduce this issue?

Install dependencies using this

```
    {
        "apps": [{
            "name": "my-app",
            "script": "./index.js",
            "cwd": "/home/www/project_root/current",
            "error_file": "/home/www/project_root/logs/app.err.log",
            "out_file": "../logs/app.out.log",
            "exec_mode":"fork_mode",
            "dependencies": [
                "pm2-logrotate"
            ]
        }]
    }
```

but there isn't a way to configure the modules.

## Supporting information

Using the latest pm2 release 2.10.0

**Comment by rohit-smpx** [comment]
I'd also like to add that the current implementation doesn't work if we are using `ecosystem.js` or anything other than `ecosystem.json`. Even though `pm2 start` can take any file as input.

**Comment by wallet77** [comment]
If you use another name than ecosystem.json you should provide the file name.
So pm2 install myfile.json

Be careful : it only takes json file extension.

**Comment by rohit-smpx** [comment]
@wallet77 That's the problem. I have my configurations in a js file, which take data from a private config file. So I can't use json file. If `pm2 start` can take js files as input, `pm2 install` should too.

**Comment by wallet77** [comment]
I think pm2 start can take a .js file cause it can run an application.
But a configuration file should be a .json

We did it as npm does !
The idea, in a next release, is to allow only configuration file with json format.

**Comment by rohit-smpx** [comment]
That does make sense. But then `package.json` is more or less static. My ecosystem file also has deploy scripts, is the start script, and the conf file.
I use the js to populate deploy scripts with secrets/keys from a private(non tracked) conf file and duplicate the app configs with different names & instances for different environments.
One of these things will no longer be required if #3122 gets some kind of solution.
I hope pm2-deploy is kept in mind going forward with the next release.

**Comment by wallet77** [comment]
Indeed, pm2-deploy and configuration / launching mechanism should be redesigned.
We need a reflection on these topics.

**Comment by rohit-smpx** [comment]
I'll be open to help, submit PRs. Better that than hack around stuff in the ecosystem file.

**Comment by nitrocode** [comment]
@rohit-smpx with all due respect, that's a completely separate issue (and a non-issue) unrelated to this ticket. Please create a separate ticket.

**Comment by vertex** [comment]
I'm also in support of configuring submodules with a config file.  It would be great to be able to have logrotate configured in the code.

**Comment by tvvignesh** [comment]
+1
Rather than having to configure logrotate separately, using a config file for the dependency will be great.

**Comment by nitrocode** [comment]
@vertex @tvvignesh for now, I used to use a `package.json` `postinstall` hook which I now changed to a `npm run configure` function that calls a shell script to configure pm2 and its modules. Reason for the `configure` script vs the `postinstall` to expedite local development and still have the pm2 configuration automated and source controlled.

**Comment by Unitech** [comment]
Great question, I will look on how we could implement that

**Comment by wallet77** [comment]
Hi there,

WIP => we will add a post-install property in ecosystem file for each dependency. So you can setup some command.

**Comment by osbornwang** [comment]
Can you tell me the final result of this problem? I did not find an example of configuring pm2-logrotate in a json file @nitrocode

**Comment by rohit-gohri** [comment]
Correct me if I am wrong but I think the whole 'dependencies' functionality was removed somewhere in between these commits, did not find any new additions in the current code related to this: https://github.com/Unitech/pm2/compare/3e3077698ca6d3741503c129585db1286a5e211e...b4575e29761fd94cdadec7f06f77f5a2ac71de52
